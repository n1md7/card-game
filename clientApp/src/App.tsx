import React from 'react';
import Authenticate from './components/Authenticate';
import Lobby from './components/Lobby';
import CreateRoom from './components/room/Create';
import JoinRoom from './components/room/Join';
import EnterRoom from './components/room/Room';
import { Redirect, Route, Switch } from 'react-router-dom';
import ProtectedRoute from './components/ProtectedRoute';

export default function App() {
  return (
    <Switch>
      <Route path="/auth/:roomId?" component={Authenticate} />
      <ProtectedRoute path="/lobby" component={Lobby} />
      <ProtectedRoute path="/create" component={CreateRoom} />
      <ProtectedRoute path="/join" component={JoinRoom} />
      <ProtectedRoute path="/G/:roomId" component={EnterRoom} />
      <Redirect to="/auth" />
    </Switch>
  );
}
