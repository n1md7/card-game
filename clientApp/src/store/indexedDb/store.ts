import { Database } from '@n1md7/indexeddb-promise';
import { getRandomInt } from '../../libs/Formulas';

export interface Users {
  id?: number;
  name: string;
  token: string;
}

export const database = new Database<Users>({
  version: 1,
  name: 'Card-Game',
  tables: [
    {
      name: 'users',
      primaryKey: {
        name: 'id',
        autoIncrement: true,
        unique: true,
      },
      initData: [
        {
          name: 'Noob' + getRandomInt(3, 3),
          token: '',
        },
      ],
      indexes: {
        name: {
          unique: false,
        },
      },
      timestamps: true,
    },
  ],
});
