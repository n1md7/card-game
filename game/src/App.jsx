import React, {useEffect, useState} from 'react';
import Cookies from "js-cookie";
import {v4 as uuidv4} from 'uuid';


import './css/app.css';

import Header from "./components/Header";
import Footer from "./components/Footer";


import 'bootstrap/dist/css/bootstrap.min.css';
import Login from "./components/Login";
import Lobby from "./components/Lobby";


function App() {

    const [auth, setAuth] = useState(false);
    const [user, setUser] = useState('');

    const logOut = () => {
        // remove cookie value
        Cookies.set('name', '');
        // reset user and trigger update
        setUser('');
        setAuth(false);
    }

    useEffect(() => {
        // get cookie value
        const name = Cookies.get('name');
        // update User state with boolean values
        setUser(name);
        setAuth(!!name);
        // and do this every auth state change
    }, [auth]);

    return (
        <div>
            {
                auth ? (
                    <>
                        <Header
                            user={user}
                            logOut={logOut}
                        />
                        <Lobby/>
                    </>
                ) :
                <Login setAuth={setAuth}/>
            }
            <Footer/>
        </div>
    );
}

export default App;
