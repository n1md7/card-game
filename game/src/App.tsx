import React from 'react';
import Grid from './components/Grid'
import './css/app.css';
import Header from "./components/Header";
import Footer from "./components/Footer";
import CreateGame from "./components/CreateGame";

const testData = {
  grid: [
    {id: 111, players: 2},
    {id: 211, players: 2},
    {id: 322, players: 1},
    {id: 347, players: 3},
    {id: 3278, players: 7}
  ]
}

function App() {
  return (
      <>
        <Header/>
        <CreateGame createGameUrl={'https://google.com'}/>
        <Grid data={testData.grid} filters={Object.fromEntries(Object.keys(testData.grid[0]).map(a => [a, '']))}/>
        <Footer/>
      </>
  );
}

export default App;
