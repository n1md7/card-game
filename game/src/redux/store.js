import {createStore, applyMiddleware} from 'redux';
import rootReducer from './reducers';
import {composeWithDevTools} from 'redux-devtools-extension';
import thunkMiddleware from 'redux-thunk';

export default createStore(
  rootReducer,
  composeWithDevTools(
    applyMiddleware(
      thunkMiddleware,
    ),
  ),
);
