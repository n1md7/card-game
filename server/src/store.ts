type UserProps = {
    id?: string;
    name?: string;
    roomId?: string;
    signUpTime?: number;
    updateTime?: number;
}

type RoomProps = {
    creator?: {
        readonly id: string;
        readonly name: string;
    };
    readonly name?: string;
    users?: string[];
    readonly size?: number;
    inRoomSize?: number;
    readonly isPublic?: boolean;
    readonly id?: string;
    readonly signUpTime?: number;
    updateTime?: number;
};

class Store {
    private users: { [id: string]: UserProps } = {/**/};
    private rooms: { [id: string]: RoomProps } = {/**/};

    public getUserById(id: string): UserProps {
        return this.users[id];
    }

    public getRoomById(id: string): RoomProps {
        return this.rooms[id];
    }

    public setRoomById(id: string, room: RoomProps): RoomProps {
        return this.rooms[id] = {
            ...room,
            id,
            inRoomSize: 1,
            signUpTime: new Date().valueOf(),
            updateTime: new Date().valueOf()
        };
    }

    public setUserById(id: string, user: UserProps): UserProps {
        // except id, signUpTime and updateTime
        // all should be passed
        return this.users[id] = {
            ...user,
            id,
            signUpTime: new Date().valueOf(),
            updateTime: new Date().valueOf()
        }
    }

    public updateRoomById(id: string, room: RoomProps): RoomProps {
        // when such id does not exist
        // it will create new record for that
        // except id and signUpTime and updateTime all can be updated
        return this.rooms[id] = {
            ...this.rooms[id],
            ...room,
            id,
            updateTime: new Date().valueOf(),
            signUpTime: this.rooms[id].signUpTime
        };
    }

    public updateUserById(id: string, user: UserProps): UserProps {
        // when such id does not exist
        // it will create new record for that
        // except id and signUpTime and updateTime all can be updated
        return this.users[id] = {
            ...this.users[id],
            ...user,
            id,
            updateTime: new Date().valueOf(),
            signUpTime: this.users[id].signUpTime
        };
    }

    public getRoomsList(){
        return this.rooms;
    }
}

const store = new Store();

export {
    store,
    RoomProps,
    UserProps
}