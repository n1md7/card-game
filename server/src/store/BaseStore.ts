export type Storage<T> = {
  [id: string]: T;
};

export default class BaseStore<T> {
  public constructor(protected storage: Storage<T> = {}) {}

  public getStorage(): Storage<T> {
    return this.storage;
  }

  public clearStorage(): void {
    this.storage = {};
  }

  public setById(id: string, item: T): T {
    return (this.storage[id] = item);
  }

  // does exactly same as setById 😁
  public updateById(id: string, item: T): T {
    return this.setById(id, item);
  }

  public getById(id: string): T | undefined {
    return this.storage[id];
  }

  public removeById(id: string): T | undefined {
    const item = this.storage[id];
    delete this.storage[id];
    return item;
  }
}
