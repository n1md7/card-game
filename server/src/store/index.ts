import PlayerStore from "./PlayerStore";
import GameStore from "./GameStore";
import UserStore from "./UserStore";
import User from "../game/User";
import Player from "../game/Player";
import Game from "../game/Game";

const playerStore = new PlayerStore();
const gameStore = new GameStore();
const userStore = new UserStore();

export {
  playerStore,
  gameStore,
  userStore,
  User,
  Player,
  Game
};
