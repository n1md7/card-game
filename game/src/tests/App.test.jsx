import React from 'react';
import { render } from '@testing-library/react';
import App from '../App';
import { Provider } from 'react-redux';
import store from '../redux/store';

test('Render app and show auth', () => {
  const { container } = render(
    <React.StrictMode>
      <Provider store={store}>
        <App />
      </Provider>
    </React.StrictMode>,
  );
  expect(container).toMatchSnapshot();
});

test.skip('Render app and trigger some events', () => {
  const { container, getAllByText } = render(
    <React.StrictMode>
      <Provider store={store}>
        <App />
      </Provider>
    </React.StrictMode>,
  );

  container.querySelector('input').value = 'Nick';
  container.querySelector('button').click();

  expect(getAllByText(/join|create/i)).toBeInTheDocument();
});
