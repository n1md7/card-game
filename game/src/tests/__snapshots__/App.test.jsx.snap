// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render app and show auth 1`] = `
<div>
  <div
    class="container my-5"
  >
    <div
      class="jumbotron"
    >
      <h1
        class="text-muted text-center"
      >
        Card game
      </h1>
      <div
        class="row justify-content-center my-5 text-center"
      >
        <div
          class="col-sm-9 col-md-4 "
        >
          <form>
            <div
              class="form-group"
            >
              <input
                autocomplete="off"
                class="form-control"
                maxlength="16"
                placeholder="Nickname"
                value=""
              />
              <small
                class="form-text text-muted"
              >
                This name will be used during the game
              </small>
            </div>
            <div
              class="row"
            >
              <div
                class="col-12"
              >
                <button
                  class="w-100 btn btn-outline-primary"
                  type="submit"
                >
                  START
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Render app and trigger some events 1`] = `
<div>
  <div
    class="container my-5"
  >
    <div
      class="jumbotron"
    >
      <h1
        class="text-muted text-center"
      >
        Card game
      </h1>
      <div
        class="row justify-content-center my-5 text-center"
      >
        <div
          class="col-sm-9 col-md-4 "
        >
          <form>
            <div
              class="form-group"
            >
              <input
                autocomplete="off"
                class="form-control"
                maxlength="16"
                placeholder="Nickname"
                value=""
              />
              <small
                class="form-text text-muted"
              >
                This name will be used during the game
              </small>
            </div>
            <div
              class="row"
            >
              <div
                class="col-12"
              >
                <button
                  class="w-100 btn btn-outline-primary"
                  type="submit"
                >
                  START
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
